modin/engines/base/frame/partition_manager.py

```diff
@@ -67,7 +67,11 @@ def column_partitions(cls, partitions):
         -------
             a list of `BaseFrameAxisPartition` objects.
         """
-        return [cls._column_partitions_class(col) for col in partitions.T]
+        if not isinstance(partitions, list):
```

Comment:
In which cases `partitions` is not a `list`?

Reply:
when we want to create column partitions from a single frame we're passing only its partitions:
```python
cls.column_partitions(frame._partitions)
```
when we want to create column partitions from multiple frames we're passing a list of partitions:
```python
cls.column_partitions([o._partitions for o in frames])
```